toolchain/check/eval.cpp

```diff
@@ -681,9 +681,58 @@ static auto PerformBuiltinBinaryIntOp(Context& context, SemIRLoc loc,
       break;
   }
 
-  bool overflow = false;
+  auto [lhs_is_signed, lhs_bit_width_id] =
+      context.sem_ir().GetIntTypeInfo(lhs.type_id);
+  llvm::APInt lhs_val = context.ints().GetAtWidth(lhs.int_id, lhs_bit_width_id);
+
   llvm::APInt result_val;
+
+  // First handle switch, which can directly use the canonical RHS and doesn't
```

Comment:
Did you mean "shift"?
```suggestion
  // First handle shift, which can directly use the canonical RHS and doesn't
```